cmd/arc/services/metamorph.go

```diff
@@ -148,14 +148,40 @@ func StartMetamorph(logger *slog.Logger) (func(), error) {
 		return nil, err
 	}
 
-	metamorphProcessor.StartLockTransactions()
+	var ctx context.Context
+	var cancel context.CancelFunc
+
+	ctx, cancel = context.WithCancel(context.Background())
+	metamorphProcessor.CancelLockTransactions = cancel
+	metamorphProcessor.WaitGroup.Add(1)
+	metamorphProcessor.StartLockTransactions(ctx)
 	time.Sleep(200 * time.Millisecond) // wait a short time so that process expired transactions will start shortly after lock transactions go routine
 
-	metamorphProcessor.StartProcessExpiredTransactions()
-	metamorphProcessor.StartRequestingSeenOnNetworkTxs()
-	metamorphProcessor.StartProcessStatusUpdatesInStorage()
-	metamorphProcessor.StartProcessMinedCallbacks()
-	err = metamorphProcessor.StartCollectStats()
+	ctx, cancel = context.WithCancel(context.Background())
+	metamorphProcessor.CancelProcessExpiredTransactions = cancel
```

Comment:
The cancel functions and wait group look like implementation details. Can't they be hidden from the client inside the processor?

Reply:
In some cases we start those functions from `cmd` directory, that's why I needed to have those exported not to change existing code. For metamorph, I did the same to have the same style.

Reply:
IMO it is more readable when all logic concerning the go routine life cycle is in processor and not in `metamorph.go`

e.g. like this


```
func (p *Processor) StartLockTransactions(ctx context.Context) {

	ctx, cancel := context.WithCancel(context.Background())
	p.cancelLockTransactions = cancel
	p.waitGroup.Add(1)

	ticker := time.NewTicker(p.lockTransactionsInterval)
	go func() {
		defer p.waitGroup.Done()
		for {
			select {
			case <-ctx.Done():
				return
			case <-ticker.C:
				expiredSince := p.now().Add(-1 * p.mapExpiryTime)
				err := p.store.SetLocked(ctx, expiredSince, loadUnminedLimit)
				if err != nil {
					p.logger.Error("Failed to set transactions locked", slog.String("err", err.Error()))
				}
			}
		}
	}()
}

```